### DISCUSSION TITLE
Material Theme Builder

### DISCUSSION THREAD

**Discussion by hestrro** [uwerat/qskinny#459] [Q&A] [answered]
Is the color algorithm of material different from that of Material Theme Builder? When I put the QRgb primary secondary of Material Theme Builder
         tertiary
        error
        neutral
     neutralVariant These 6 colors have different effects when applied to QSkinny

**Comment by uwerat** [comment, marked as answer]
- the algo is explained here: https://material.io/blog/science-of-color-design
- the implementation itself is code from Google, that has been translated by me to C++ ( https://github.com/uwerat/qskinny/blob/master/src/common/QskHctColor.cpp )
- the tonal values used for the skin can be found in the QskMaterial3Theme constructor ( https://github.com/uwerat/qskinny/blob/master/designsystems/material3/QskMaterial3Skin.cpp#L1397 )

a)
Please note that our background is related to Embedded/Linux platforms. We rarely use Windows and never use Android environments in our daily workflows. Maybe you can contribute to QSkinny by testing and evaluating how far it is for the Android platform ?

b)
It looks like you have a Chinese background, right. We once implemented a Pinyin text prediction for our virtual keyboard but in the end we have no clue if it is correct or makes any sense at all ?

**Comment by hestrro** [comment]
Yes, I'm Chinese and I'm happy to contribute

**Reply by uwerat** [reply]
Excellent. Could you please contact me by email ( the one you find on the project page ).

**Reply by hestrro** [reply]
I have sent a message to support@qskinny.org, the name is He Yang
